pkg/manifests/types.go

```diff
@@ -258,6 +258,9 @@ type ThanosQuerierConfig struct {
 	// The possible values are `error`, `warn`, `info`, and `debug`.
 	// The default value is `info`.
 	LogLevel string `json:"logLevel,omitempty"`
+	// A Boolean flag that enables setting CORS headers.
+	// The default value is `false`.
+	EnableCORS bool `json:"enableCORS,omitempty"`
```

Comment:
Just a thought: 
should we use `*bool` so that we can determine (in future) without changing the API if the value has been explicitly set by user?

Reply:
Do note it is only a `suggestion` , feel free to ignore :) 

> Is there a convention? 

I don't think there is one. In this specific case, I think we may want to change the default behaviour later (i.e. if user hasn't explicitly set this field) and making it a pointer (like k8s api types do) helps with that.

> Do we need to apply the default/get rid of nil like in here 

I would says so ...  k8s pointer package provides utils for this. 

```
import "k8s.io/utils/pointer"

pointer.BoolPtr(false)
```

Reply:
> I don't think there is one. In this specific case, I think we may want to change the default behaviour later (i.e. if user hasn't explicitly set this field) and making it a pointer (like k8s api types do) helps with that.

My bad, I meant a convention for "applying the default/getting rid of nil like in the example". ;)

Thanks, I'll take a look at the package.

Reply:
I haven't tested it, so it would be nice to know if it is possible to convert this to pointer later and detect if the field was set 🤔 . Have you tried it already?

Reply:
It looks straightforward to me, converting it to a pointer will still allow the user to provide the same values, we'd only need to deal with nil (here we'd know the user didn't provide a value and we could set the value it points to to false to avoid breaking the API), I just think it's not worth all the extra code for now as detecting missing values is not needed for now.

Reply:
+1 If we really need to know if a user set this or not, a pointer is good. But since this effectively passes through a boolean cli flag `*bool` seems unnecessary here.
Or is there a specific use case you have in mind Sunil?

Reply:
> since this effectively passes through a boolean cli flag *bool seems unnecessary here

Based on the test added in this PR, I thought that isn't the case. I.E. this config will  be stored in the configmap.

Reply:
Sorry, is one of the added implies that the code detects/should detect missing values?

Reply:
For the record, we resolved this. My concern was with mashalling and unmarshalling which is never the case since we only unmarshall.